ui/core/components/suggest_reforges_action.tsx

```diff
@@ -1164,7 +1169,15 @@ export class ReforgeOptimizer {
 			const filteredGemDataForColor = new Array<GemData>();
 
 			for (const gem of allGemsOfColor) {
-				if (gem.requiredProfession > 0 || gem.name.includes('Perfect') || !gemMatchesSocket(gem, socketColor)) {
+				if (
+					(gem.requiredProfession == Profession.Jewelcrafting && !this.player.hasProfession(Profession.Jewelcrafting)) ||
```

Comment:
Minor performance optimization: pre-cache
```
const isJc := this.player.hasProfession(Profession.Jewelcrafting)
```
at the start of the `buildGemOptions()` method.